xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -2423,8 +2431,11 @@ public String getSkinFile(String filename, boolean forceSkinAction, XWikiContext
             }
 
             // Look for a resource file
-            if (resourceExists("/resources/" + filename)) {
+            String resourceFilePath = "/resources/" + filename;
+            if (resourceExists(resourceFilePath)) {
                 URL url = urlf.createResourceURL(filename, forceSkinAction, context);
+                url = new URIBuilder(url.toURI()).addParameters(getResourceURLParameters(resourceFilePath)).build()
```

Comment:
Would probably be better for performances and more consistent to add a createResourceURL taking a `String querystring` like it's the case for others XWikiURLFactory methods.